lib/ui/screens/login_screen/login_screen.dart

```diff
@@ -327,8 +348,16 @@ class _LoginScreenState extends ConsumerState<_LoginScreen> {
         MaterialPageRoute(builder: (_) => const FilterScreen()),
       );
     } on RPCError catch (e) {
+      if (e.code == RPCError.twoFactorRequired) {
+        requireTwoFactor = true;
+      }
       ref.read(loginStateProvider.notifier).state = ref.read(loginStateProvider).copyWith(
-            message: e.code == RPCError.authenticationFailed ? "Falsches Passwort" : e.message,
+            message: switch (e.code) {
+              RPCError.authenticationFailed => "Falsches Passwort",
+              RPCError.twoFactorRequired => "2-Faktor-Token benötigt",
```

Comment:
Should there be an error message in this case? I'd rather skip this by using `return` in the if-statement above.